src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1896,6 +1908,48 @@ export class SavedObjectsRepository {
     return body;
   }
 
+  /**
+   * Returns a generator to help page through large sets of saved objects.
+   *
+   * The generator wraps calls to `SavedObjects.find` and iterates over
+   * multiple pages of results using `_pit` and `search_after`. This will
+   * open a new Point In Time (PIT), and continue paging until a set of
+   * results is received that's smaller than the designated `perPage`.
+   *
+   * Once you have retrieved all of the results you need, it is recommended
+   * to call `close()` to clean up the PIT and prevent Elasticsearch from
+   * consuming resources unnecessarily. This will automatically be done for
+   * you if you reach the last page of results.
+   *
+   * @example
+   * ```ts
+   * const findOptions: SavedObjectsCreatePointInTimeFinderOptions = {
+   *   type: 'visualization',
+   *   search: 'foo*',
+   *   perPage: 100,
+   * };
+   *
+   * const finder = savedObjectsClient.createPointInTimeFinder(findOptions);
+   *
+   * const responses: SavedObjectFindResponse[] = [];
+   * for await (const response of finder.find()) {
+   *   responses.push(...response);
+   *   if (doneSearching) {
+   *     await finder.close();
+   *   }
+   * }
+   * ```
+   */
+  createPointInTimeFinder(
+    findOptions: SavedObjectsPointInTimeFinderOptions,
+    dependencies: Omit<SavedObjectsPointInTimeFinderDependencies, 'logger'>
+  ) {
+    return new PointInTimeFinder(findOptions, {
+      logger: this._logger,
+      ...dependencies,
+    });
+  }
```

Comment:
Is there any particular reason to make the `dependencies` parameter required for the SOR but not for the SOC?

It would be more consistent to do this:
```
  createPointInTimeFinder(
    findOptions: SavedObjectsPointInTimeFinderOptions,
    dependencies?: Omit<SavedObjectsPointInTimeFinderDependencies, 'logger'>
  ) {
    return new PointInTimeFinder(findOptions, {
      logger: this._logger,
      find: this.find.bind(this),
      openPointInTimeForType: this.openPointInTimeForType.bind(this),
      closePointInTime: this.closePointInTime.bind(this),
      // Include dependencies last so that SO client wrappers have their settings applied.
      ...dependencies,
    });
  }
```